### PR TITLE
chore: upgrade babel/typescript to support latest features

### PR DESCRIPTION
**Pull request by scurker**
General cleanup of typescript/babel dependencies. Removed a few unused babel plugins, but there were some issues with `import type...` on #1167 that I was trying to resolve here separately.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -128,7 +127,7 @@
     "style-loader": "^0.19.0",
     "terser-webpack-plugin": "^5.3.6",
     "ts-node": "^10.9.1",
-    "typescript": "^3.7.5",
+    "typescript": "~5.0.0",
```
Comment (scurker) [inline package.json:130]: prettier yells at you if it's anything greater than 5.1.0 🤷

File: packages/react/.babelrc [hunk 2]
```diff
@@ -11,7 +11,6 @@
         }
       ],
       "transform-export-extensions",
-      "@babel/plugin-proposal-class-properties",
```
Comment (scurker) [inline packages/react/.babelrc:14]: now included as part of preset-env

File: packages/react/package.json [hunk 3]
```diff
@@ -34,18 +34,15 @@
     "tslib": "^2.4.0"
   },
   "devDependencies": {
-    "@babel/cli": "^7.8.4",
```
Comment (scurker) [inline packages/react/package.json:37]: unused

File: packages/react/src/components/IconButton/index.tsx [hunk 4]
```diff
@@ -115,7 +115,6 @@ IconButton.propTypes = {
   as: PropTypes.elementType,
   // @ts-expect-error
   icon: PropTypes.string.isRequired,
-  // @ts-expect-error
```
Comment (scurker) [inline packages/react/src/components/IconButton/index.tsx:118]: apparently none of these error anymore

File: package.json [hunk 5]
```diff
@@ -23,7 +23,8 @@
     "release": "./scripts/release.sh"
   },
   "prettier": {
-    "singleQuote": true
+    "singleQuote": true,
+    "trailingComma": "none"
```
Comment (scurker) [inline package.json:27]: I don't really care that much, but this keeps the existing behavior that was present before the upgrade to keep things consistent

File: package.json [hunk 6]
```diff
@@ -43,14 +44,12 @@
   },
   "devDependencies": {
     "@axe-core/puppeteer": "^4.3.1",
-    "@babel/core": "^7.4.3",
-    "@babel/plugin-proposal-class-properties": "^7.4.0",
-    "@babel/plugin-proposal-export-default-from": "^7.2.0",
-    "@babel/preset-env": "^7.8.4",
-    "@babel/preset-react": "^7.0.0",
-    "@babel/preset-stage-0": "^7.0.0",
```
Comment (scurker) [inline package.json:51]: not really needed anymore

File: packages/react/src/components/IconButton/index.tsx (hunk 4 above)
Comment (dequejosie) [reply packages/react/src/components/IconButton/index.tsx:118]: The others still error?

**Comment by dequejosie** [comment]
Looks good to me!

**Review by dequejosie — APPROVED** [review]

**Pull request merged**